internal/dag/gatewayapi_processor.go

```diff
@@ -1556,6 +1842,92 @@ func (p *GatewayAPIProcessor) clusterRoutes(routeNamespace string, matchConditio
 			TimeoutPolicy:         ClusterTimeoutPolicy{ConnectTimeout: p.ConnectTimeout},
 		})
 	}
+	return clusters, totalWeight, true
+}
+
+// GRPCClusters builds clusters from backendRef.
+func (p *GatewayAPIProcessor) GRPCClusters(routeNamespace string, backendRefs []gatewayapi_v1alpha2.GRPCBackendRef, routeAccessor *status.RouteParentStatusUpdate, protocolType gatewayapi_v1beta1.ProtocolType) ([]*Cluster, uint32, bool) {
```

Comment:
Nit, let's lowercase the method name here:

```suggestion
func (p *GatewayAPIProcessor) grpcClusters(routeNamespace string, backendRefs []gatewayapi_v1alpha2.GRPCBackendRef, routeAccessor *status.RouteParentStatusUpdate, protocolType gatewayapi_v1beta1.ProtocolType) ([]*Cluster, uint32, bool) {
```